checkov/terraform/graph_builder/local_graph.py

```diff
@@ -172,90 +177,114 @@ def _build_edges(self) -> None:
         aliases = self._get_aliases()
         resources_types = self.get_resources_types_in_graph()
         for origin_node_index, vertex in enumerate(self.vertices):
-            for attribute_key, attribute_value in vertex.attributes.items():
-                if attribute_key in reserved_attribute_names or attribute_has_nested_attributes(
+            self._build_edges_for_vertex(origin_node_index, vertex, aliases, resources_types)
+
+    def _build_edges_for_vertex(self, origin_node_index: int, vertex: TerraformBlock, aliases: Dict[str, Dict[str, BlockType]], resources_types: List[str], cross_variable_edges: bool = False):
+        for attribute_key, attribute_value in vertex.attributes.items():
+            if attribute_key in reserved_attribute_names or attribute_has_nested_attributes(
                     attribute_key, vertex.attributes
-                ):
-                    continue
-                referenced_vertices = get_referenced_vertices_in_value(
-                    value=attribute_value,
-                    aliases=aliases,
-                    resources_types=resources_types,
-                )
-                for vertex_reference in referenced_vertices:
-                    # for certain blocks such as data and resource, the block name is composed from several parts.
-                    # the purpose of the loop is to avoid not finding the node if the name has several parts
-                    sub_values = [remove_index_pattern_from_str(sub_value) for sub_value in vertex_reference.sub_parts]
-                    for i, _ in enumerate(sub_values):
-                        reference_name = join_trimmed_strings(char_to_join=".", str_lst=sub_values, num_to_trim=i)
-                        if vertex.module_dependency:
+            ):
+                continue
+            referenced_vertices = get_referenced_vertices_in_value(
+                value=attribute_value,
+                aliases=aliases,
+                resources_types=resources_types,
+            )
+            for vertex_reference in referenced_vertices:
+                # for certain blocks such as data and resource, the block name is composed from several parts.
+                # the purpose of the loop is to avoid not finding the node if the name has several parts
+                sub_values = [remove_index_pattern_from_str(sub_value) for sub_value in vertex_reference.sub_parts]
+                for i in range(sub_values):
```

Comment:
```suggestion
                for i in range(len(sub_values)):
```
that fixes your integration issues